### ISSUE TITLE
Abstract class should not be instantiable

### ISSUE THREAD

**Issue by cmp0xff** [astral-sh/ty#4306]
### Summary

https://play.ty.dev/2a6425a1-b0c8-4a57-ab35-831ce16c0157

```py
from typing import ClassVar
from abc import ABC, abstractmethod

class A(ABC):
    m: ClassVar[str]  # abstract class variable
    @abstractmethod

    def f(self) -> None:
        pass

A()  # should fail type checking
```

### Version

playground
